contracts/bridge/x/v4/RegistrationV4.sol

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2024
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {CoreV4} from "./CoreV4.sol";
+
+contract RegistrationV4 {
+    CoreV4 public immutable imx;
+
+    constructor(address payable _imx) {
+        imx = CoreV4(_imx);
+    }
+
+    function getVersion() external view returns (string memory) {
+        return imx.VERSION();
+    }
+
+    function isRegistered(uint256 starkKey) public view returns (bool) {
+        return imx.getEthKey(starkKey) != address(0);
+    }
+
+    function registerAndWithdrawAll(address ethKey, uint256 starkKey, bytes calldata signature, uint256 assetType)
+        external
+    {
+        if (!isRegistered(starkKey)) {
+            imx.registerEthAddress(ethKey, starkKey, signature);
+        }
+        withdrawAll(uint160(ethKey), starkKey, assetType);
+    }
+
+    function withdrawAll(uint256 ethKey, uint256 starkKey, uint256 assetType) public {
```

Comment:
There is no check that the ethKey relates to the starkKey. Could this be used to attack the system?

Reply:
explained on slack, but I'll also reply here:

>it doesn't really matter if the sender or the ethKey is the owner of that given starkKey because the withdrawal will be done to what starkex contract has on their side and not to what we tell them. So we don't have to add protective measures here, once a user can call the starkex contract directly, it should be adding the safeguards, not us.